File: OpenWeather/Converters/TemperatureConverter.vb
Imports System.Globalization

Public Class TemperatureConverter
    Implements IMultiValueConverter

    Public Function Convert(values() As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IMultiValueConverter.Convert
        Dim icon = CStr(values(0))
        Dim dayTemperature = CDbl(values(1))
        Dim nightTemperature = CDbl(values(2))

        If icon Is Nothing Then Return Binding.DoNothing

        If icon.ElementAt(2) = "d" Then Return dayTemperature Else Return nightTemperature
    End Function

    Public Function ConvertBack(value As Object, targetTypes() As Type, parameter As Object, culture As CultureInfo) As Object() Implements IMultiValueConverter.ConvertBack
        Return Binding.DoNothing
    End Function
End Class

File: OpenWeatherCS/Converters/TemperatureConverter.cs
using System;
using System.Globalization;
using System.Linq;
using System.Windows.Data;

namespace OpenWeatherCS.Converters
{
    public class TemperatureConverter : IMultiValueConverter
    {
        public object Convert(object[] values, Type targetType, object parameter, CultureInfo culture)
        {
            var icon = (string) values[0];
            var dayTemperature = (double) values[1];
            var nightTemperature = (double) values[2];

            if (icon == null) return Binding.DoNothing;

            return (icon.ElementAt(2) == 'd') ? dayTemperature : nightTemperature;            
        }

        public object[] ConvertBack(object value, Type[] targetTypes, object parameter, CultureInfo culture)
        {
            return (object[]) Binding.DoNothing;
        }
    }
}
